### PR TITLE
Better ASA deploy error/warning handle

### LINKED ISSUE #697: Mocha printing "ASA file not defined"
## Describe the bug

On running mocha tests, I am getting "ASA file not defined" printed on the console. This happens after I create the runtime.

_I have a stateful smart contract .teal files in the assets folder_ but **no ASA files**

#### Expected Behavior

Nothing printed when there are not ASA files in assets folder

#### Current Behavior

Printing "ASA file not defined" when there are no ASA files in the assets folder.

image

## Possible Solution

This issue possibly arises from this line called in the constructor of `Runtime`

https://github.com/scale-it/algo-builder/blob/326d6d34a7279dc0efc32cbcab0fc9ad4161c293/packages/runtime/src/runtime.ts#L77

and the warning is printed here:
https://github.com/scale-it/algo-builder/blob/326d6d34a7279dc0efc32cbcab0fc9ad4161c293/packages/runtime/src/lib/files.ts#L88

## Steps to Reproduce

1. Create a test case with before hook
2. Add `Runtime([])` call in the before hook
3. The warning will get printed in the console when you run the test using `npm test`

```js
    before(function () {
        // runs once before the first test in this block
        runtime = new Runtime([]);
    });
```
### Environment

- OS: Windows
- Node.js version: 16.15.0
- algob version: NA

### PR DESCRIPTION
**Pull request by vuvoth**
Closes #697 

## Proposed Changes

- Throw error when deploy asa from asa.yaml but asa.yml not defined
- Throw error when deploy asa but asa definition not found in asa.yml.
- Not show any warning

### CODE REVIEW

File: packages/runtime/src/errors/errors-list.ts [hunk 1]
```diff
@@ -698,6 +698,19 @@ const runtimeAsaErrors = {
...
+		number: 1511,
+		message: "ASA file is undefined.",
+		title: "ASA file is undefined.",
+		description:
+			"ASA file(asa.yml) is undefined however you tried to read ASA defintion from this file.",
+	},
+	ASA_DEFINITION_NO_FOUND_IN_ASA_FILE: {
+		number: 1511,
+		message: "ASA defition not found in asa file(asa.yml)",
+		title: "ASA definition not found in asa file(asa.yml)",
```
Comment (vuvoth) [inline packages/runtime/src/errors/errors-list.ts:711]:
```suggestion
		message: "ASA defition not found in asa file(asa.yaml)",
		title: "ASA definition not found in asa file(asa.yaml)",
```

File: packages/runtime/src/errors/errors-list.ts [hunk 2]
```diff
@@ -698,6 +698,19 @@ const runtimeAsaErrors = {
 		title: "Cannot close asset ID in allocating account",
 		description: "Asset holding of Asset creator cannot be closed to other account",
 	},
+	ASA_FILE_IS_UNDEFINED: {
+		number: 1511,
+		message: "ASA file is undefined.",
+		title: "ASA file is undefined.",
+		description:
+			"ASA file(asa.yml) is undefined however you tried to read ASA defintion from this file.",
```
Comment (sczembor) [inline packages/runtime/src/errors/errors-list.ts:706]:
```suggestion
			"Attempt to read an undefined ASA file (asa.yml)",
```

File: packages/runtime/test/integration/deploy-asa.ts [hunk 3]
```diff
@@ -35,4 +37,27 @@ describe("Deploy ASA with mutiple opt-in accounts", function () {
...
+			() =>
+				runtime.deployASA("asa-invalid", {
+					creator: { ...john.account, name: "john" },
+				}),
+			RUNTIME_ERRORS.ASA.ASA_DEFINITION_NO_FOUND_IN_ASA_FILE
+		);
+	});
+
+	describe("ASA file is undefinded", function () {
+		useFixture("loop"); // project don't have asa-file
```
Comment (sczembor) [inline packages/runtime/test/integration/deploy-asa.ts:52]:
```suggestion
		useFixture("loop"); // project does not have an asa-file
```

File: packages/runtime/test/integration/deploy-asa.ts [hunk 4]
```diff
@@ -35,4 +37,27 @@ describe("Deploy ASA with mutiple opt-in accounts", function () {
...
+		it("Should fail b/c we tried to get asa from asa file for deploy", () => {
```
Comment (sczembor) [inline packages/runtime/test/integration/deploy-asa.ts:53]: I think we should you a convention as above. `Should throw an error when...`

File: packages/runtime/src/errors/errors-list.ts [hunk 5]
```diff
@@ -698,6 +698,19 @@ const runtimeAsaErrors = {
 		title: "Cannot close asset ID in allocating account",
 		description: "Asset holding of Asset creator cannot be closed to other account",
 	},
+	ASA_FILE_IS_UNDEFINED: {
+		number: 1511,
+		message: "ASA file is undefined.",
+		title: "ASA file is undefined.",
+		description:
+			"Attempt to read an undefined ASA file (asa.yml)",
```
Comment (vuvoth) [inline packages/runtime/src/errors/errors-list.ts:706]:
```suggestion
			"Attempt to read an undefined ASA file (asa.yaml)",
```

**Comment by PabloLION** [comment]
Dear @vuvoth,
My TS engine throwing error to me that "This condition will always return 'false' since JavaScript compares objects by reference, not value.ts(2839)" on line 245.

https://github.com/scale-it/algo-builder/blob/7c986f0b24774179aaf348e65e3c30e94b1e5c91/packages/runtime/src/ctx.ts#L244-L252

Can you confirm if it's implemented correctly?
Should I create a new issue?

**Pull request merged**